dqn.py

```diff
@@ -128,14 +153,15 @@ def train(self, batch_size, sample_memory=False):
         states = states.view(batch_size, -1)
 
         # clipping of rewards
-        rewards[rewards > 1.0] = 1.0
-        rewards[rewards < -1.0] = -1.0
+        rewards = torch.clamp(rewards, -1.0, 1.0)
 
         q_val = self.compute_q_vals(states, actions)
         with torch.no_grad():  # Don't compute gradient info for the target (semi-gradient)
             target = self.compute_targets(batch_size, rewards, next_states, dones)
 
-        loss = F.smooth_l1_loss(q_val, target)
+        # loss = F.smooth_l1_loss(q_val, target)
```

Comment:
I put the smooth version as that was suggested in the lab by the TAs. However, in the paper they use MSE loss which is L2 loss rather than L1.

Reply:
You're right! I read this line incorrectly.
> Because the absolute value loss function `|x|` has a derivative of -1 for all negative values of `x` and a derivative of 1 for all positive values of `x`, clipping the squared error to be between -1 and 1 corresponds to using an absolute value loss function for errors outside of the (-1,1) interval.

I will change it to being MSE + clipping.